phoenix-core/src/it/java/org/apache/phoenix/end2end/JsonArrayElementsFunctionIT.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.end2end;
+
+import org.apache.phoenix.exception.SQLExceptionCode;
+import org.apache.phoenix.schema.types.PhoenixArray;
+import org.apache.phoenix.util.PropertiesUtil;
+import org.junit.Test;
+
+import java.sql.*;
+import java.util.Properties;
+
+import static org.apache.phoenix.util.TestUtil.TEST_PROPERTIES;
+import static org.junit.Assert.*;
+
+/**
+ * End to end test for {@link org.apache.phoenix.expression.function.JsonArrayElementsFunction}.
+ *
+ */
+public class JsonArrayElementsFunctionIT extends BaseHBaseManagedTimeIT {
+
+    @Test
+    public void testJsonArrayElementsWithSameType() throws Exception {
+        Connection conn = getConnection();
+
+        try {
+            String json = "[25.343,36.763,37.56,386.63]";
+            String pk = "valueOne";
+
+            populateJsonTable(conn, json, pk);
+
+            String selectQuery = "SELECT json_array_elements(col1) FROM testJson WHERE pk = 'valueOne'";
```

Comment:
@JamesRTaylor   the postgress implementation returns a SETOF json, while this returns a varchar array. Do you think this is OK?